### PR TITLE
Add skipRegistry option to settings.json

### PR DESCRIPTION
**Pull request by andre2007**
In a company scenario you want to specify that only the company internal dub registry is used and not any public registry (security issues...). While it was already possible to define registryUrls in settings.json, disabling the public dub registries (code.dlang.org, ....) wasn't possible. This pull request allows to set the skipRegistry option in settings.json.

### CODE REVIEW

**Comment by andre2007** [comment]
That's true. But I wonder whether adding new constructor would introduce too much complexity. I have another suggestion. I extract the whole package supplier logic from the current constructor into a new function which has a import argument the skipPackageSupplier value and returns the list of determined package suppliers. In the existing constructor I can then decide whether to call this function with the skip package supplier value from the constructor import argument or the default value from settings.json. This would also ease writing a unit test for the new functionality. What do you think?

**Comment by WebFreak001** [comment]
0 cost testable abstraction which makes the code more readable always sounds good

**Comment by andre2007** [comment]
I did the changes and added a unittest. Please have a look.

**Comment by andre2007** [comment]
@WebFreak001 could you have a second look? Thanks

File: source/dub/dub.d [hunk 1]
```diff
@@ -167,6 +167,47 @@ class Dub {
...
+	/** Get the list of package suppliers.
+
+		Params:
+			additional_package_suppliers = A list of package suppliers to try
+				before the suppliers found in the configurations files and the
+				`defaultPackageSuppliers`.
+			skip_registry = Can be used to skip using the configured package
+				suppliers, as well as the default suppliers.
+	*/
+	private PackageSupplier[] getPackageSuppliers(PackageSupplier[] additional_package_suppliers, SkipPackageSuppliers skip_registry)
```
Comment (WebFreak001) [inline source/dub/dub.d]: if this is well documented and unittested I think it would make sense to make this public

**Comment by WebFreak001** [comment]
looks good, can you just make that a public symbol (and the ditto function), it seems like a function that could be useful if used as a library. (For example show suppliers in IDE)

File: source/dub/dub.d (hunk 1 above)
Comment (andre2007) [reply source/dub/dub.d]: done :)

File: source/dub/dub.d [hunk 2]
```diff
@@ -167,6 +167,47 @@ class Dub {
...
+	}
+
+	unittest
+	{
+		scope (exit) environment.remove("DUB_REGISTRY");
+		auto dub = new Dub(".", null, SkipPackageSuppliers.configured);
+		assert(dub.m_packageSuppliers.length == 0);
+		environment["DUB_REGISTRY"] = "http://example.com/";
+		dub = new Dub(".", null, SkipPackageSuppliers.configured);
+		logInfo("%s", dub.m_packageSuppliers);
```
Comment (wilzbach) [inline source/dub/dub.d]: I would probably remove this to avoid polluting the output when `dub test` is run.

Comment (andre2007) [reply source/dub/dub.d]: fixed :)

**Pull request merged**